internal/dataplane/sendconfig/sendconfig.go

```diff
@@ -58,8 +58,11 @@ func PerformUpdate(
 		return oldSHA, fmt.Errorf("failed to generate SHA for target content: %w", err)
 	}
 
-	// disable optimization if reverse sync is enabled
-	if !config.EnableReverseSync {
+	// Disable the optimization by checking if the configuration is same as the last applied configuration if:
+	// - reverse sync is enabled
+	// - or the update is performed by a fallback process because in DBmode, when applying fails, some entities are applied successfully, then the Kong gateway may be in a "partial success" state.
+	// REVIEW: Only disable the check on fallback in DBmode?
```

Comment:
I _think_ that we can disable this check in both DB mode only. When working with dbless it's a simple hash to compare whether there's need apply new config.

So the condition below would need to change.